### PR TITLE
Feature: Listing

### PR DESCRIPTION
**Pull request by lrongyi**

### CODE REVIEW

File: src/main/java/seedu/address/model/listing/Listing.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+ * Represents a Listing in the real estate application.
+ * Guarantees: all fields are present and not null, and buyers is initialized to an empty set.
+ */
+public class Listing {
+    private final Price price;
+    private final Size size;
+    private final Location location;
+    private final Address address;
+    private final Person seller;
+    private final Set<Person> buyers;
```
Comment (muhdbhz) [inline src/main/java/seedu/address/model/listing/Listing.java:22]: Note to change to Buyer once buyer is implemented

File: src/main/java/seedu/address/model/listing/Listing.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+/**
+ * Represents a Listing in the real estate application.
+ * Guarantees: all fields are present and not null, and buyers is initialized to an empty set.
+ */
+public class Listing {
+    private final Price price;
+    private final Size size;
+    private final Location location;
+    private final Address address;
+    private final Person seller;
```
Comment (muhdbhz) [inline src/main/java/seedu/address/model/listing/Listing.java:21]: Note to change to Seller once seller is implemented

File: src/main/java/seedu/address/model/listing/Listing.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    /**
+     * Constructs a {@code Listing}.
+     *
+     * @param address  Address of the listing.
+     * @param price    Price of the listing.
+     * @param size     Size of the listing in square meters.
+     * @param location Location of the listing, represented by a {@code Location} enum.
+     * @param seller   Seller of the listing, represented by a {@code Person}.
+     */
+    public Listing(Address address, Price price, Size size, Location location, Person seller) {
```
Comment (muhdbhz) [inline src/main/java/seedu/address/model/listing/Listing.java:33]: Same as above

**Review by muhdbhz — APPROVED** [review]
LGTM! Changes mentioned above are for future commits

**Comment by wang-h-z** [comment]
This comment pertains the most recent commit made by me.

Known Issues: 
1. Listings are not saved properly in the listings.json file when added via the CLI.
2. UI needs to be added to show the listings properly for user testing.
3. seedu.address.model.listing.Address is redundant and should be deleted, it is currently causing conflicts with seedu.address.model.person.Address.
4. Add parsing rules for Listing and its respective classes
5. Incorporate buyer and seller
6. Deserialization of seedu.address.model.listing.Area is not working due to String processing.

**Pull request merged**